lib/reverse_sentence.rb

```diff
@@ -1,4 +1,51 @@
-# A method to reverse the words in a sentence, in place.
+#A method to reverse the words in a sentence, in place.
+# The time complexity:
+# The first iteration is done when the entire sentence is reversed.
+# The second iteration when the start and end indices are found.
+# The third time the iterations are (each_word.length)/2 times for reversing the word.
+# Since the iterations happen sequentially, the time complexity is O(n)
+#
+# The space complexity is O(1) as the sentence is reversed in place.
```

Comment:
And because the amount of auxiliary storage you need does not change as the size of the input changes. 

Yes, you're reversing the string in place. What's better is you are not using a temporary, auxiliary array or string to hold all the characters in the input and then placing them back in the original string. You're doing it using only constant amount storage to hold starting and ending indices, a temporary character to perform the swap etc. - the memory needed does not change if there are 50 characters or 5,000 characters in the input string.